agent/lib/src/external_api/model.rs

```diff
@@ -0,0 +1,139 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    pub filename: String,
+    pub content: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct CodeEntry {
+    pub files: Vec<FileModel>,
+    pub script: Vec<String>, // All commands to execute at startup
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    Status,
+    Request,
+    Response,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Code {
+    Run,
+    Ok,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct StatusMessage {
+    pub r#type: Type,
+    pub code: Code,
+}
+
+impl StatusMessage {
+    pub fn new() -> StatusMessage {
+        StatusMessage {
+            r#type: Type::Status,
+            code: Code::Ok,
```

Comment:
I had to look at the main function to understand that you're using `StatusMessage` only for notifying the host that the agent is *ready*. Using a `Ready` status code may be better then. And `new` should take a status code as it's input.